shared/src/event_handling.rs

```diff
@@ -170,11 +170,7 @@ where
     S: EventStoring<C::Event> + Send + Sync,
 {
     async fn run_maintenance(&self) -> Result<()> {
-        self.lock()
-            .await
-            .update_events()
-            .await
-            .context("event update error")
+        self.lock().await.update_events().await
```

Comment:
This is the 🔑  change to propagate internal errors better

Reply:
Isn't this only removing one layer of context? Wouldn't it without the change below now just say 

`shared::maintenance: failed with: replace - failed to convert events`

which still wouldn't tell you much about what's happening?

Reply:
I think the other key change is :point_down: where we use the debug instead of display formatter to print the error, thereby printing the entire error chain and not the most recent context (as you also noted).

Reply:
Looking forward to running into these errors again! Will be ready for them now.